src/adagrams.js

```diff
@@ -1,15 +1,133 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const numTiles = 10;
+  const lettersInHand = [];
+  const currentLetterPool = [...listOfLetters]
+  for (let i = 0; i < numTiles; i++) {
+    // 98 letters, i want index from 0 to 97 inclusive
+    let randIndex = Math.floor(Math.random() * currentLetterPool.length);
+    let randLetter = currentLetterPool[randIndex];
+    lettersInHand.push(randLetter);
+    currentLetterPool.splice(randIndex, 1)
+  }
+  return lettersInHand
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const letterMap = lettersInHandToDict(lettersInHand);
+  input = input.toUpperCase();
+  for (const char of input) {
+    // doesn't have char in map -> false
+    // has char in map
+      // if value > 0 -> -1
+      // if value = 0 -> false
+    if (!letterMap.has(char)) {
+      return false;
+    } else if (letterMap.has(char)) {
+      if (letterMap.get(char) > 0) {
+        letterMap.set(char, letterMap.get(char) - 1);
+      } else {
+        return false;
+      }
+    }
+  }
+  // if it hasn't return false at this point, return true
+  return true;
 };
 
+const lettersInHandToDict = lettersInHand => {
+  const letterMap = new Map();
+  for (const char of lettersInHand) {
+    // char already exists in the map, get value and += 1
+    if (letterMap.has(char)) {
+      letterMap.set(char, letterMap.get(char) + 1);
+    } else { // char not in dict yet, create it with value 1
+      letterMap.set(char, 1);
+    }
+  }
+  return letterMap
+}
+
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  word = word.toUpperCase();
+
+  if (! word) {
+    return 0;
+  }
+
+  let score = 0;
+  for (const char of word) {
+    score += letterValues[char]
+  }
+  if (word.length >= 7) {
+    score += 8
+  }
+  return score
 };
 
 export const highestScoreFrom = (words) => {
-  // Implement this method for wave 4
+  let maxScore = 0;
+  let maxWord = '';
+
+  for (const word of words) {
+    const score = scoreWord(word)
+    if (score > maxScore) {
+      maxScore = score;
+      maxWord = word;
+    }
+    else if (score === maxScore) {
+      if (maxWord.length === 10) {
+        // keep the original word, since it came first
+        continue
+      } else if (word.length === 10) {
+        // make word the new maxWord
+        maxWord = word;
+      } else if (word.length < maxWord.length) {
+        // make word the new maxWord 
+        maxWord = word;
+      } else {
+        // keep maxWord to maxWord
+        continue
+      }
+    }
+  }
+
+  return { 
+    score: maxScore,
+    word: maxWord };
```

Comment:
When we break up objects across multiple lines, the closing curly brace should be on its own line.
```suggestion
  return { 
    score: maxScore,
    word: maxWord 
  };
```